adam-core/src/main/scala/org/bdgenomics/adam/hbase/HBaseFunctions.scala

```diff
@@ -0,0 +1,587 @@
+
+    val tableDescriptor = new HTableDescriptor(TableName.valueOf(hbaseTableName))
+    tableDescriptor.addFamily(new HColumnDescriptor("g".getBytes()).setCompressionType(Algorithm.GZ)
+      .setDataBlockEncoding(DataBlockEncoding.FAST_DIFF)
+      .setMaxVersions(1))
+
+    if (splitsFileName.isDefined) {
+
+      // splits defines a set of redefined HBase region splits based on an array of Array[Byte] indicating the
+      // split points
+      val splits: Array[Array[Byte]] = Source.fromFile(splitsFileName.get).getLines.toArray
+        .map(x => x.split(" "))
+        .map(g => Bytes.toBytes(g(0) + "_"
+          + String.format("%10s", g(1)).replace(' ', '0')))
+
+      dao.admin.createTable(tableDescriptor, splits)
+    } else dao.admin.createTable(tableDescriptor)
+
+    val hbaseTableNameMeta = hbaseTableName + "_meta"
+
+    val tableDescriptorMeta: HTableDescriptor = new HTableDescriptor(TableName.valueOf(hbaseTableNameMeta))
+    tableDescriptorMeta.addFamily(new HColumnDescriptor("meta".getBytes())
+      .setCompressionType(Algorithm.GZ)
+      .setDataBlockEncoding(DataBlockEncoding.FAST_DIFF)
+      .setMaxVersions(1))
+    dao.admin.createTable(tableDescriptorMeta)
+  }
+
+  /**
+   *
+   * @param dao HBase Data Access Object
+   * @param vcRdd VariantContextRDD from which to save data to HBase
+   * @param hbaseTableName HBase table name
+   * @param sequenceDictionaryId user defined sequence Dionctionary Id to use when saving
+   * @param saveSequenceDictionary toggle saving of Sequence Dictioanry, set to false when relevant Sequence
+   *                               Dictionary has already been saved to HBase in a previous load operation
+   * @param partitions - number N of partitions in which to repartition data, if not defined data is not repartitioned
+   * @param stagingFolder - user defined location (on HDFS) to which to save temporary staging files
+   */
+
+  def saveVariantContextRDDToHBaseBulk(dao: HBaseSparkDAO,
+                                       vcRdd: VariantContextRDD,
+                                       hbaseTableName: String,
+                                       sequenceDictionaryId: String,
+                                       stagingFolder: String,
+                                       saveSequenceDictionary: Boolean = true,
+                                       partitions: Option[Int] = None) = {
+
+    saveSampleMetadataToHBase(dao, hbaseTableName + "_meta", vcRdd.samples)
+
+    if (saveSequenceDictionary) saveSequenceDictionaryToHBase(dao,
+      hbaseTableName + "_meta",
+      vcRdd.sequences,
+      sequenceDictionaryId)
+
+    val rdd = if (partitions.isDefined) vcRdd.rdd.repartition(partitions.get)
+    else vcRdd.rdd
+
+    val genodata: RDD[(Array[Byte], List[(String, Array[Byte])])] = rdd.mapPartitions((iterator) => {
+      val genotypebaos: java.io.ByteArrayOutputStream = new ByteArrayOutputStream()
```

Comment:
I think we have similar code in the `rdd` package, so perhaps we can refactor it out?